### ISSUE TITLE
after you deployed last version 1.9.1 

### ISSUE THREAD

**Issue by mohamedaboelmagd** [code-farmz/ng-otp-input#103]
my angular app was 12 and I used already 1.8.1
after [code-farmz] deployed version 1.9.1
app didn't accept to deploy
I tried to upgrade it to 1.9.1 to fix this issue, still issue exist

./node_modules/ng-otp-input/fesm2015/ng-otp-input.mjs - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js):
TypeError: Cannot create property 'message' on string '/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/ng-otp-input/fesm2015/ng-otp-input.mjs: This application depends upon a library published using Angular version 15.2.9, which requires Angular version 14.0.0 or newer to work correctly.
Consider upgrading your application to use a more recent version of Angular.
  37 | }
  38 | /** @nocollapse */ KeysPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
> 39 | /** @nocollapse */ KeysPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, name: "keys" });
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  40 | i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, decorators: [{
  41 |             type: Pipe,
  42 |             args: [{'
    at run (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/@babel/core/lib/transformation/index.js:37:15)
    at run.next (<anonymous>)
    at Function.transform (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/@babel/core/lib/transform.js:25:41)
    at transform.next (<anonymous>)
    at step (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:261:32)
    at /home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:223:11)
    at /home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:37:40

Error: node_modules/ng-otp-input/lib/components/ng-otp-input/ng-otp-input.component.d.ts:33:18 - error TS2314: Generic type 'ɵɵComponentDeclaration' requires 7 type argument(s).

33     static ɵcmp: i0.ɵɵComponentDeclaration<NgOtpInputComponent, "ng-otp-input", never, { "config": "config"; "formCtrl": "formCtrl"; }, { "onInputChange": "onInputChange"; }, never, never, false, never>;
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/ng-otp-input/lib/pipes/keys.pipe.d.ts:6:19 - error TS2314: Generic type 'ɵɵPipeDeclaration' requires 2 type argument(s).

6     static ɵpipe: i0.ɵɵPipeDeclaration<KeysPipe, "keys", false>;
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by code-farmz** [comment]
Try adding ```"skipLibCheck": true``` under compilerOptions section of tsconfig.json or to move back to ```1.8.1``` change package configuration under your package.json to "ng-otp-input": "1.8.1" and run ```npm i``` so it use the specific version instead of the latest one

<img width="245" alt="image" src="https://github.com/code-farmz/ng-otp-input/assets/19927847/735798de-f9f1-45b6-8e46-432c3c89c632">

**Comment by mohamedaboelmagd** [comment]
still not working, I do your solution @code-farmz  

./node_modules/ng-otp-input/fesm2015/ng-otp-input.mjs - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js):
TypeError: Cannot create property 'message' on string '/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/ng-otp-input/fesm2015/ng-otp-input.mjs: This application depends upon a library published using Angular version 15.2.9, which requires Angular version 14.0.0 or newer to work correctly.
Consider upgrading your application to use a more recent version of Angular.
  37 | }
  38 | /** @nocollapse */ KeysPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
> 39 | /** @nocollapse */ KeysPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, name: "keys" });
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  40 | i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: KeysPipe, decorators: [{
  41 |             type: Pipe,
  42 |             args: [{'
    at run (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/@babel/core/lib/transformation/index.js:37:15)
    at run.next (<anonymous>)
    at Function.transform (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/@babel/core/lib/transform.js:25:41)
    at transform.next (<anonymous>)
    at step (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:261:32)
    at /home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:223:11)
    at /home/mohamed/Desktop/careerograph/talent-hub-web/node_modules/gensync/index.js:37:40

**Comment by code-farmz** [comment]
then you will have to use a specific package version i.e. v1.8.1

**Comment by mohamedaboelmagd** [comment]
I am already use it

**Comment by mohamedaboelmagd** [comment]
I run those two commands while deploying to any environment to make it works
- npm install
- npm install ng-otp-input@1.8.1 -S
please fix it as it's not necessary to do it

**Comment by mohamedaboelmagd** [comment]
I fix it by replacing "ng-otp-input": "^1.8.1", to     "ng-otp-input": "~1.8.1",

**Comment by code-farmz** [comment]
yes that's what I recommend above you can more read about versioning [here](https://stackoverflow.com/a/49193722/5621827)
